### ISSUE TITLE
Weird fitting when upper bound frequency is higher than 100 Hz. 

### ISSUE THREAD

**Issue by Drvinc** [fooof-tools/fooof#310]
The PSD was generated using welch. 2s hann window, 50% overlap. fs = 2048 Hz.

The fit looks good when freq_range = [3, 105]
![image](https://github.com/fooof-tools/fooof/assets/12715598/f560d8f9-2f0a-49f1-96bf-343226495df4)

However, when the upper bound is extended freq_range = [3, 400]
![image](https://github.com/fooof-tools/fooof/assets/12715598/a33f6fe8-3942-4f7f-86a9-c0760a61f7dc)

Am I doing anything wrong?

**Comment by TomDonoghue** [comment]
Hey @Drvinc - sorry for a slow reply here, I forgot to reply to this back when it was posted. 

What seems to be happening here is that above about 100 Hz, your data is hitting a noise floor (as possibly was filtered), and so it's flattening out. The model expects a continuous 1/f decay, and so cannot properly capture this plateau - basically the shape of the data across this broad range is different from what the model expects / is designed for, and so it does it's best, but ends up fitting weird shape you see above. If you check the plots in log-log space, the 1/f line, and how this can't capture the noise floor, will probably be clearer. 

Long story short, in cases like this, a single fit across the whole range won't work. My guess is that there probably isn't anything to meaningfully capture above ~100 Hz here (it looks like it's decaying to a noise floor) - but if you did want to measure something there, you would have to try something like a piecewise fit across different frequency ranges.

**Comment by TomDonoghue** [comment]
Hey @Drvinc - since this is a a data question with no clear action item in the code I'm going to close this now - but feel free to reply to the issue if you have any follow up questions, etc!
